app/src/main/kotlin/com/wire/android/ui/home/HomeDialogs.kt

```diff
@@ -29,43 +29,74 @@ import com.wire.android.R
 import com.wire.android.ui.common.WireDialog
 import com.wire.android.ui.common.WireDialogButtonProperties
 import com.wire.android.ui.common.WireDialogButtonType
+import com.wire.android.ui.home.messagecomposer.state.SelfDeletionDuration
 import com.wire.android.util.CustomTabsHelper
+import kotlin.time.Duration.Companion.seconds
+import kotlin.time.DurationUnit
+import kotlin.time.toDuration
 
 @Composable
 fun FileRestrictionDialog(
-    featureFlagState: FeatureFlagState,
+    isFileSharingEnabled: Boolean,
     hideDialogStatus: () -> Unit,
 ) {
-    if (featureFlagState.showFileSharingDialog) {
-        val text: String = if (featureFlagState.isFileSharingEnabledState) {
-            stringResource(id = R.string.sharing_files_enabled)
-        } else {
-            stringResource(id = R.string.sharing_files_disabled)
-        }
+    val text: String = stringResource(id = if (isFileSharingEnabled) R.string.sharing_files_enabled else R.string.sharing_files_disabled)
 
-        WireDialog(
-            title = stringResource(id = R.string.team_settings_changed),
-            text = text,
-            onDismiss = hideDialogStatus,
-            optionButton1Properties = WireDialogButtonProperties(
-                onClick = hideDialogStatus,
-                text = stringResource(id = R.string.label_ok),
-                type = WireDialogButtonType.Primary,
-            )
+    WireDialog(
+        title = stringResource(id = R.string.team_settings_changed),
+        text = text,
+        onDismiss = hideDialogStatus,
+        optionButton1Properties = WireDialogButtonProperties(
+            onClick = hideDialogStatus,
+            text = stringResource(id = R.string.label_ok),
+            type = WireDialogButtonType.Primary,
+        )
+    )
+}
+
+@Composable
+fun SelfDeletingMessagesDialog(
+    isSelfDeletingMessagesEnabled: Boolean,
+    enforcedTimeout: SelfDeletionDuration,
+    hideDialogStatus: () -> Unit,
+) {
+    val text: String = if (isSelfDeletingMessagesEnabled) {
+        val formattedTimeout = formatEnforcedTimeout(enforcedTimeout)
+        if (enforcedTimeout == SelfDeletionDuration.None)
+            stringResource(id = R.string.self_deleting_messages_team_setting_enabled)
+        else stringResource(id = R.string.self_deleting_messages_team_setting_enabled_enforced_timeout, formattedTimeout)
+    } else stringResource(id = R.string.self_deleting_messages_team_setting_disabled)
+
+    WireDialog(
+        title = stringResource(id = R.string.team_settings_changed),
+        text = text,
+        onDismiss = hideDialogStatus,
+        optionButton1Properties = WireDialogButtonProperties(
+            onClick = hideDialogStatus,
+            text = stringResource(id = R.string.label_ok),
+            type = WireDialogButtonType.Primary,
         )
-    }
+    )
+}
+
+@Composable
```

Comment:
no need for this we have this : 
```

@Suppress("MagicNumber")
enum class SelfDeletionDuration(val value: Duration?, val label: UIText) {
    None(null, UIText.StringResource(R.string.label_off)),
    TenSeconds(10.seconds, UIText.PluralResource(R.plurals.seconds_label, 10, 10)),
    FiveMinutes(5.minutes, UIText.PluralResource(R.plurals.minutes_label, 5, 5)),
    OneHour(1.hours, UIText.PluralResource(R.plurals.hours_label, 1, 1)),
    OneDay(1.days, UIText.PluralResource(R.plurals.days_label, 1, 1)),
    OneWeek(7.days, UIText.PluralResource(R.plurals.days_label, 7, 7)),
    FourWeeks(28.days, UIText.PluralResource(R.plurals.weeks_label, 4, 4))
}

```

Reply:
I added this function before you finished your PR, so now I can remove it. Thx for the reminder.